### ISSUE TITLE
Lifecycle changes to improve startup performance

### ISSUE THREAD

**Issue by DesignSpaceBot** [adobe-photoshop/spaces-design#448]
This PR splits the `onStartup` action lifecycle method into `beforeStartup` and `afterStartup` methods. Startup is considered complete after all the former actions have finished executed. And only after they're all complete do we execute the latter actions. 

This also document initialization in two parts: initializing the active document in `beforeStartup`, and initializing the inactive documents later in `afterStartup`. Building up the system font tables is also delayed until `afterStartup`, as that can take at least a second or two. (The side effect of this is that the `Type`, if active on startup, will momentarily show the current typeface as "missing." We might want to soften that a little.)  

Finally, this also change the timing of the UI fade-in from 500ms to 100ms. I did this because the long fade-in became choppy when when doing the heavy `afterStartup` work of loading all the inactive documents. This improves the choppiness, and also just feels a little more nice and brisk to me. I'm sure @timriot will have additional thoughts.

In summary, this should significantly improve startup time when there is more than one document open, and slightly improve startup time if there is only one document open due to the delayed system font loading. This addresses #355 as written, though we may still want another issue to track overall startup time.

-- @iwehrman at [2015-01-22T20:19:07Z](https://github.com/adobe-photoshop/spaces-design-archive/pull/448)
